src/client/java/teammates/client/scripts/sql/VerifyCourseEntityAttributes.java

```diff
@@ -0,0 +1,163 @@
+                        : sqlEntity.getDeletedAt().equals(datastoreEntity.getDeletedAt());
+    }
+
+    private boolean verifySectionChain(teammates.storage.sqlentity.Course sqlEntity) {
+        // Get old and new students
+        List<CourseStudent> oldStudents = ofy().load().type(CourseStudent.class).filter("courseId", sqlEntity.getId())
+                .list();
+        List<Student> newStudents = getNewStudents(sqlEntity.getId());
+
+        // Group students by section
+        Map<String, List<CourseStudent>> sectionToOldStuMap = oldStudents.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getSectionName));
+        Map<String, List<Student>> sectionToNewStuMap = newStudents.stream()
+                .collect(Collectors.groupingBy(Student::getSectionName));
+
+        List<Section> newSection = sqlEntity.getSections();
+
+        boolean isSectionsCountEqual = newSection.size() != sectionToOldStuMap.size()
+                || newSection.size() != sectionToNewStuMap.size();
+        if (!isSectionsCountEqual) {
+            return false;
+        }
+
+        return newSection.stream().allMatch(section -> {
+            List<CourseStudent> oldSectionStudents = sectionToOldStuMap.get(section.getName());
+            List<Student> newSectionStudents = sectionToNewStuMap.get(section.getName());
+
+            // If sectionStudent is null, then section is not present in sql
+            boolean isSectionNamePresent = oldSectionStudents != null && newSectionStudents != null;
+            if (!isSectionNamePresent) {
+                return false;
+            }
+
+            // Group students by team
+            Map<String, List<CourseStudent>> teamNameToOldStuMap = oldSectionStudents.stream()
+                    .collect(Collectors.groupingBy(CourseStudent::getTeamName));
+            Map<String, List<Student>> teamNameToNewStuMap = newSectionStudents.stream()
+                    .collect(Collectors.groupingBy(Student::getTeamName));
+            return verifyTeams(section, teamNameToOldStuMap, teamNameToNewStuMap);
+        });
+
+    }
+
+    private boolean verifyTeams(Section newSection,
+            Map<String, List<CourseStudent>> teamNameToOldStuMap, Map<String, List<Student>> teamNameToNewStuMap) {
+
+        List<Team> newTeam = newSection.getTeams();
+
+        boolean isTeamCountEqual = newTeam.size() != teamNameToNewStuMap.size()
+                || newTeam.size() != teamNameToOldStuMap.size();
+        if (!isTeamCountEqual) {
+            return false;
+        }
+
+        return newTeam.stream().allMatch(team -> {
+            List<CourseStudent> oldTeamStudents = teamNameToOldStuMap.get(team.getName());
+            List<Student> newTeamStudents = teamNameToNewStuMap.get(team.getName());
+
+            // If teamStudents is null, then team is not present in sql
+            boolean isTeamNamePresent = oldTeamStudents != null && newTeamStudents != null;
```

Comment:
Please correct me if I'm wrong, should this be that the team is not present in datastore instead? Since we're streaming the SQL team.  Same for "If sectionStudent is null, then section is not present in sql"

Reply:
I made the comments clearer